components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/JWTAccessTokenOIDCClaimsHandler.java

```diff
@@ -91,63 +109,569 @@ public JWTClaimsSet handleCustomClaims(JWTClaimsSet.Builder builder, OAuthAuthzR
           Handling the user attributes for the access token. There is no requirement of the consent
           to manage user attributes for the access token.
          */
-        String clientId = request.getAuthorizationReqDTO().getConsumerKey();
-        String spTenantDomain = getServiceProviderTenantDomain(request);
-        AuthenticatedUser authenticatedUser = request.getAuthorizationReqDTO().getUser();
+        Map<String, Object> claims = getUserClaimsInOIDCDialect(request);
+        return setClaimsToJwtClaimSet(builder, claims);
+    }
+
+    /**
+     * Get user claims in OIDC dialect.
+     *
+     * @param requestMsgCtx OAuthTokenReqMessageContext
+     * @return User claims in OIDC dialect
+     * @throws IdentityOAuth2Exception IdentityOAuth2Exception
+     */
+    private Map<String, Object> getUserClaimsInOIDCDialect(OAuthTokenReqMessageContext requestMsgCtx)
+            throws IdentityOAuth2Exception {
+
+        Map<String, Object> userClaimsInOIDCDialect;
 
-        Map<String, Object> claims = getAccessTokenUserClaims(authenticatedUser, clientId, spTenantDomain);
-        if (claims == null || claims.isEmpty()) {
-            return builder.build();
+        Map<ClaimMapping, String> userAttributes = getCachedUserAttributes(requestMsgCtx, false);
+        if ((userAttributes.isEmpty() || isOrganizationSwitchGrantType(requestMsgCtx))
```

Comment:
AuthorisationGrantCache has only the OIDC requested claims. If a user is a local user, and if he has some OIDC requested attributes, this will be false. And it will go to the federation scenario. 

Any reason for not checking for local user to enter into this if condition?